src/components/UserSelector.tsx

```diff
@@ -1,40 +1,92 @@
-import React from 'react';
+import { useEffect, useState, useRef } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+
+type Props = {
+  users: User[];
+  selectedUser: User | null;
+  setSelectedUser: (user: User | null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  selectedUser,
+  setSelectedUser,
+}) => {
+  const userSelect = useRef<HTMLDivElement | null>(null);
+  const dropdownMenu = useRef<HTMLDivElement | null>(null);
+
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+
+  const handleOutsideClick = (event: MouseEvent) => {
+    if (
+      dropdownMenu.current &&
+      userSelect.current &&
+      !dropdownMenu.current.contains(event.target as Node) &&
+      !userSelect.current.contains(event.target as Node)
+    ) {
+      setIsDropdownActive(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleOutsideClick);
+
+    return () => {
+      document.removeEventListener('mousedown', handleOutsideClick);
+    };
+  }, []);
+
+  const handleDropdownClick = (user: User) => {
+    if (!selectedUser || user.id !== selectedUser.id) {
+      setSelectedUser(user);
+    }
+
+    setIsDropdownActive(!isDropdownActive);
```

Comment:
The dropdown is toggled after selecting a user, which might not be the intended behavior. Consider ensuring that the dropdown closes after a user is selected.